gemorris / Human Experience Edits (Envisioning/Discovery) (#868)
microsoft/code-with-engineering-playbook: docs/machine-learning/ml-problem-formulation-envisioning.md

# Discovery
##### Envisioning and Problem Formulation

Before beginning a data science investigation, we need to define a problem statement which the data science team can explore; this problem statement can have a significant influence on whether the project is likely to be successful.

## Discovery goals

The main goals of the discovery process are:

> *	Establish existing understanding of the problem domain and the underlying business objectives

Reviewer: The information is great, so thank you for the contribution. 

In general, however, this text is a bit heavy to read - can we rewrite this in a bit more "Plain English" (see [http://www.plainenglish.co.uk/how-to-write-in-plain-english.html](http://www.plainenglish.co.uk/how-to-write-in-plain-english.html) for tips) to make sure our intentions come across correctly? 

Examples of a few potential rewordings

- Understand the problem domain, and business objectives
- (? Not completely sure what this sentence means, but perhaps) State the limits of the problem?
- Explain how potential solutions may interface with existing code, and how we can measure the performance of the suggested solutions.
- Determine what data is available, and the origin and quality of the data. Take special care to understand if the data may be biased.
- ...

Similar with the rest of the document.

Author: @TessFerrandez Noted, I will be able to get to this around the holidays

The following questions can help guide discussion in understanding the stakeholders' perspectives:

*	Who are the people affected by the outcomes of the proposed system? (These may not be just end users)
*	What is the current practice related to the business problem?
*	What are the local initiatives related to the current problem statement? How can we learn from, leverage, and build on those efforts?
*	What is the current solution? How does it perform?
*	What opportunities or untapped potential exists in the problem space? 
*	What are the frictions experienced by people who are affected by the problem today?
*	What are the most consistent or pressing challenges related to the problem?
*	What is the state of the data used to build the solution?
*	How does the end user or SME envision the solution?

## Envisioning Phase

During the envisioning phase and related efforts, the following guides may prove useful for directing the discussion and activities, as well as identifying and coordinating with key resources on the project. Many of these points are taken directly, or adapted from, [[1]](#references) and [[2]](#references).

> ### Problem Framing & Hypothesis Generation

Reviewer 2: Hypothesis generation seem to me like an activity that should be done at the next step in the project. Should it be part of the envisioning phase? We usually perform hypothesis driven analysis/development once we start data and business understanding (i.e., the problem is already formulated).

Author: Part of Discovery, which for HX Research is Descriptive and Generative research of the business and human context, would be "generating theory to test" e.g. hypothesis. It would not necessarily include the testing of those theories, but would at least provide them as part of the "findings" of a discovery.

Would love to collaborate on this and get more of your feedback in early December if you are around then.

1. Define the objective in business terms, if there is a lack of consensus re-visit previous work undertaken to achieve this objective definition, and consider any new information or perspectives brought forward.
2. Review the understanding of the problem context that was gathered through research and sensemaking activities prior to the workshop, including but not limited to:

>    0.	History of the problem

Reviewer 2: ```suggestion
   1.	History of the problem
```
List isn't rendered correctly. Best practice is to have all items as 1. (the number would automatically increment during rendering)

   1.	Human context and impact
   2.	Social and organizational context
   3.	Potential impact of intended outcomes
3. What are the current solutions/workarounds (if any). Is it possible to solve manually? How?
4. Review the work that has been done in this area so far.

> 5. Identify if this solution need to fit into existing business IT systems.

Reviewer 2: ```suggestion
5. Identify if this solution needs to fit into existing business IT/production systems.
```

6. Identify the existing or proposed business processes that this solution must fit within.
7. Determine how performance should be measured.
8. Confirm the performance measure is aligned with the business objective.
9. Define the minimum performance needed to reach the business objective.
10. Identify any known constraints around non-functional requirements that would have to be considered. (e.g., computation times)
11. Provide a framing for this problem based on understanding from discovery activities and participant input (supervised/unsupervised, online/offline, etc.)

> 12. Identify if human expertise is available.

Reviewer 2: ```suggestion
12. Confirm that human expertise is available.
```
Just a suggestion. Building such capabilities with no SMEs would likely result in a wrong implementation.
